smartsim/_core/launcher_/slurm/slurm_launcher.py

```diff
@@ -0,0 +1,334 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import collections
+import os
+import os.path
+import subprocess as sp
+import time
+import typing as t
+
+from typing_extensions import TypeAlias
+
+from smartsim._core.config import CONFIG
+from smartsim._core.launcher_.slurm import slurm_commands as commands
+from smartsim._core.launcher_.slurm import slurm_parser as parser
+from smartsim._core.utils import helpers, launcher
+from smartsim.error import errors
+from smartsim.log import get_logger
+from smartsim.status import JobStatus
+from smartsim.types import LaunchedJobID
+
+if t.TYPE_CHECKING:
+    from smartsim.experiment import Experiment
+
+
+logger = get_logger(__name__)
+
+
+class SrunCommand:
+    """A type to inject job important information such as the job name and ID
```

Comment:
Consider revising for verbosity and awkward wording...
`Produces a properly formatted srun command from raw job information.`

The docstring as-is appears to say that SrunCommand is all about tracking but that appears to be not totally true. Is there another SrunCommand that _doesn't_ allow tracking? What about:

`Produces a properly formatted srun command from raw job information. Ensures the job status can be tracked using the supplied name.`

Reply:
Love it!! Stealing that verbatim!